doc/release/0.17.0-notes.rst

```diff
@@ -29,6 +29,9 @@ Deprecated features
 Backwards incompatible changes
 ==============================
 
+The functions ``trim1`` and ``trimboth`` now sort elements before trimming.
+This is backwards incompatible but it made no sense to trim unsorted input.
```

Comment:
the larger the array the faster partition is than sort as its a linear scaling instead of a linarithmic one, for even for 1000 elements I still get a factor 2 better performance.

I don't expect of a function named `trim` to sort the data, its not called `trim_and_sort`

Also note that `trim_mean` now sorts/partitions twice as it calls `trimboth`